zcash_client_backend/src/proto/service.rs

```diff
@@ -284,17 +284,6 @@ pub mod compact_tx_streamer_client {
     pub struct CompactTxStreamerClient<T> {
         inner: tonic::client::Grpc<T>,
     }
-    impl CompactTxStreamerClient<tonic::transport::Channel> {
-        /// Attempt to create a new client by connecting to a given endpoint.
-        pub async fn connect<D>(dst: D) -> Result<Self, tonic::transport::Error>
-        where
-            D: TryInto<tonic::transport::Endpoint>,
-            D::Error: Into<StdError>,
-        {
-            let conn = tonic::transport::Endpoint::new(dst)?.connect().await?;
-            Ok(Self::new(conn))
-        }
-    }
```

Comment:
I can confirm that we're depending on this`impl`.

Reply:
> What we ideally want is for this impl to not be present when the target is WASM, but available otherwise. 

Curious to understand why you think that. Seems like pulling in the provided this should be irrespective of what `target_arch` you're in. Maybe the `transport` feature will be supported by `wasm32` at some point (obviously not), but still. I'm of the opinion the less stuff you need to pull into `zcash_client_backend/src/proto.rs`, the better.